### ISSUE TITLE
Bug: "Bus error" after 180000 blocks indexed on Raspberry PI 4 B

### ISSUE THREAD

**Issue by agyx** [romanz/electrs#889]
**Describe the bug**

 - "Bus error" after 180000 blocks indexed.

**Electrs version**

 - 0.9.13 (rust-rocksdb: 0.15.0-e3)

**To Reproduce**

 - remove the db directory
 - start electrs: `electrs --conf config.toml` 
 - wait until 180000 blocks are indexed
 - electrs quits with "Bus error"

**Expected behavior**

 - no crash expected.

**Configuration**

 - Config file:
 ```
cookie_file = "/mnt/data/bitcoin/data/.cookie"
daemon_rpc_addr = "127.0.0.1:8332"
daemon_p2p_addr = "127.0.0.1:8333"
db_dir = "/mnt/data/electrs/data/db"
network = "bitcoin"
electrum_rpc_addr = "127.0.0.1:50001"
log_filters = "INFO"
 ```

 - environment variables:  nothing specific to dev or electrs
 - arguments: `--conf config.toml`
 - rustc --version: `rustc 1.48.0 (471d696c7 2020-12-10)`
- cargo --version: `cargo 1.46.0`

**System compiling AND running electrs**

 - Compiled natively with: `RUSTFLAGS="-C link-args=-latomic" cargo build --locked --release`
 - Deployment method: manual
 - OS: Raspbian GNU/Linux 11 (bullseye) / Linux 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr  3 17:24:16 BST 2023 aarch64 GNU/Linux
 - Hardware: Raspberry PI 4 B 4GB, raspbian in ro, mounted SSD 2GB for data

**Logs in release mode**
```
...
[2023-06-04T19:58:22.140Z INFO  electrs::index] indexing 2000 blocks: [176001..178000]
[2023-06-04T19:59:16.291Z INFO  electrs::chain] chain updated: tip=00000000000009eae2697a7aaf57e730b707b9f4530449c16d924d534d41f297, height=178000
[2023-06-04T19:59:16.483Z INFO  electrs::index] indexing 2000 blocks: [178001..180000]
[2023-06-04T20:00:45.652Z INFO  electrs::chain] chain updated: tip=00000000000004ff83b6c10460b239ef4a6aa320e5fffd6c7bcedefa8c78593c, height=180000
[2023-06-04T20:00:45.844Z INFO  electrs::index] indexing 2000 blocks: [180001..182000]
Bus error
bash$
```

**Debug information with gdb**
```
Thread 10 "rocksdb:high0" received signal SIGBUS, Bus error.
[Switching to Thread 0xf59fe340 (LWP 11940)]
ZSTD_compressBlock_doubleFast_generic (mls=4, srcSize=<optimized out>, src=<optimized out>, cctx=0xf5c14008) at zstd/lib/compress/zstd_double_fast.c:131
131                     hashLong[ZSTD_hashPtr(ip, hBitsL, 8)] = (U32)(ip-base);
```

**Electrum client**

 - no client needed

**Other comments**

- v0.8.12 runs fine (rust-rocksdb: 0.12.2)

**Comment by romanz** [comment]
https://github.com/romanz/electrs/issues/226 looks relevant.

**Comment by romanz** [comment]
Have you installed Rust from Raspbian repositories, or via rustup.rs?

**Comment by agyx** [comment]
I once tried `rustup` install but did not put much effort on it at that time. I tried again harder and this config actually works (ie: node fully sync and usable):
- triplet: `armv7-unknown-linux-gnueabihf`
- version: `beta`

I remember that the `buster` + `apt install cargo` config was working fine on `0.9.x` but this is not true anymore with `bullseye`. I tried using the `cargo` package from `buster` but `electrs` does not compile (because of a compile error with `tiny_http`).
